### PR TITLE
feat: add synonym setting in plugin setting and sync with typenses

### PR DESCRIPTION
**Pull request by ridwanarifandi**
image

image

### CODE REVIEW

**Review by jv-lopez — APPROVED** [review]

File: app/Settings/SynonymSettings.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+		if ( self::$instance === null ) {
+			self::$instance = new self( 'wooless_synonym_settings_options' );
+		}
+
+		return self::$instance;
+	}
+
+	public function register_hooks() {
+		// add_filter( 'blaze_wooless_additional_site_info', array( $this, 'register_additional_site_info' ), 10, 1 );
+	}
```
Comment (jv-lopez) [inline app/Settings/SynonymSettings.php:22]: is this being used?

File: views/synonym-management.php [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+		const appendTemplate = () => {
+			// count how many rows are there
+			let totalRows = $('.blaze-wooless-synonym-row').length;
+
+			console.log({ totalRows })
+
+			if (totalRows > 0) {
+				console.log($('#blaze-wooless-synonym-row-holder').find('.blaze-wooless-synonym-row').last())
+				// find last .blaze-wooles-synonym-row
```
Comment (jv-lopez) [inline views/synonym-management.php:123]: remove  debug lines

File: views/synonym-management.php [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+				totalRows = parseInt($('#blaze-wooless-synonym-row-holder').find('.blaze-wooless-synonym-row').last().data('row')) + 1;
+			}
+
+			console.log({ totalRows })
```
Comment (jv-lopez) [inline views/synonym-management.php:127]: remove debug lines

File: views/synonym-management.php [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+			<div class="blaze-wooless-synonym-row__actions">
+				<button class="blaze-wooless-synonym-row__actions__remove">
+					<span class="dashicons dashicons-trash"></span>
+				</button>
+			</div>
+		</div>
+	</div>
+</script>
+<script type="text/javascript">
+	(function ($) {
```
Comment (jv-lopez) [inline views/synonym-management.php:112]: why not add this js into our main js file? is there a problem if we separate the script?

**Review by jv-lopez — CHANGES_REQUESTED** [review]

File: views/synonym-management.php (hunk 4 above)
Comment (ridwanarifandi) [reply views/synonym-management.php:112]: i think we can keep it inline, since the script is only used on this page only

**Review by jv-lopez — APPROVED** [review]

**Pull request merged**